core/scheduler.h

```diff
@@ -260,9 +181,107 @@ class Scheduler final {
 
   double ns_per_cycle_;
 
+ private:
   DISALLOW_COPY_AND_ASSIGN(Scheduler);
 };
 
+// The default "fast" scheduler, which picks the first leaf that the TC tree
+// gives it and runs the corresponding task.
+template <typename CallableTask>
+class FastScheduler : public Scheduler<CallableTask> {
+ public:
+  explicit FastScheduler(TrafficClass *root = nullptr) : Scheduler<CallableTask>(root) {}
+
+  virtual ~FastScheduler() {}
+
+  // Runs the scheduler loop forever.
+  void ScheduleLoop() override {
+    uint64_t now;
+    // How many rounds to go before we do accounting.
+    const uint64_t accounting_mask = 0xff;
+    static_assert(((accounting_mask + 1) & accounting_mask) == 0,
+                  "Accounting mask must be (2^n)-1");
+
+    this->checkpoint_ = now = rdtsc();
+
+    // The main scheduling, running, accounting loop.
+    for (uint64_t round = 0;; ++round) {
+      // Periodic check, to mitigate expensive operations.
+      if ((round & accounting_mask) == 0) {
+        if (ctx.is_pause_requested()) {
+          if (ctx.BlockWorker()) {
+            break;
+          }
+        }
+      }
+
+      ScheduleOnce();
+    }
+  }
+
+  // Runs the scheduler once.
+  void ScheduleOnce() __attribute__((always_inline)) {
```

Comment:
We don't need `__attribute__((always_inline))` here; the attribute is mostly irrelevant in C++ context. As long as a member function is defined in the class body it will be inlined according to the spec.